syntax = "proto3"; // 指定使用 Protocol Buffers v3 语法
package krpc; // 定义命名空间


// 消息结构：【header长度】+ 【 header（服务+函数+参数长度）】+ 【函数参数 args】

// 这里定义的就是 header 部分，RpcHeader 是 RPC 请求中附带的 “消息头” ，用于告诉服务端：
//    调用哪个服务类；
//    调用哪个方法；
//    后面参数数据的长度是多少


// 服务端解析这个头部后，就能反射式地调用对应的服务方法（通过 service_name + method_name）
// 再从网络流中提取args_size长度的参数内容


message rpcHeader // 定义消息结构，描述RPC调用的元消息
{
    bytes service_name = 1; // 服务名
    bytes method_name = 2;  // 方法名
    uint32 args_size = 3;   // 参数序列化后的大小
}